### PR TITLE
Finishing Feature installment plans

### PR DESCRIPTION
**Pull request by PhilDL**
- Fix object name "intallment_plan"
- Refactor mapper InstallmentPlan data to have APIResource object like Payment
- Adding Schedule APIResource and initializer to map list to list of Schedule(s)
- Add tests notifications treat
- Fix tests to use new APIResource objects

### CODE REVIEW

File: payplug/resources.py [hunk 1]
```diff
@@ -397,7 +405,37 @@ class InstallmentPlan(APIResource, VerifiableAPIResource, ReconstituableAPIResou
...
+        Initialize a resource.
+        Default behavior is just to set all the attributes. You may want to override this.
+
+        :param resource_attributes: The resource attributes
+        """
+
+        schedules = []
+        if resource_attributes.get('schedule', []):
+            for schedule in resource_attributes['schedule']:
+                schedules.append(InstallmentPlan.Schedule(**schedule))
```
Comment (calops) [inline payplug/resources.py]:
```suggestion
        for schedule in resource_attributes.get('schedule', []):
            schedules.append(InstallmentPlan.Schedule(**schedule))
```

File: payplug/resources.py [hunk 2]
```diff
@@ -62,13 +63,15 @@ def factory(data):
             exceptions.UnkownAPIResource when it's impossible to reconstruct the APIResource from its data.
         """
         if 'object' not in data:
-            raise exceptions.UnknownAPIResource('Missing `object` key in resource.')
+            raise exceptions.UnknownAPIResource(
+                'Missing `object` key in resource.')
```
Comment (calops) [inline payplug/resources.py]: We usually allow 120 characters columns in our codebase, could you revert these aesthetic changes?

**Review by calops — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution!

File: payplug/resources.py (hunk 1 above)
Comment (PhilDL) [reply payplug/resources.py]: Thank you for your feeback, done.

File: payplug/resources.py (hunk 2 above)
Comment (PhilDL) [reply payplug/resources.py]: I reverted all code formatting changes

**Comment by PhilDL** [comment]
I've reverted stylistic changes and squashed the commits.

**Review by calops — APPROVED** [review]

**Pull request merged**